### PR TITLE
Create check new api token

### PR DESCRIPTION
**Pull request by Uarhodesian**
check create api-token

### CODE REVIEW

File: features/api_token_settings.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+Feature: Checking account settings
+
```
Comment (timbortnik) [inline features/api_token_settings.feature]: Can we make this feature independent? (not assume we are already logged in)

File: features/environment_secret.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: UTF-8 -*-
 import os
 
-HIPCHAT_LOGIN = os.environ.get("HCLOGIN", "tim+test@bortnik.org")
-HIPCHAT_PASS = os.environ.get("HCPASS", "ntcn987654321")
+HIPCHAT_LOGIN = os.environ.get("HCLOGIN", "danilyuk@bigmir.net")
+HIPCHAT_PASS = os.environ.get("HCPASS", "vjhzxjrntcn")
```
Comment (timbortnik) [inline features/environment_secret.py]: Any password exposed to open repository should be changed ASAP

File: features/steps/step_selenium_api_token.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+# @mark.steps
+# ----------------------------------------------------------------------------
+# STEPS:
+# ----------------------------------------------------------------------------
+from behave import given, when, then
+import time
+
+@given('we are on Hipchat Home Page')
+def step_impl(context):
```
Comment (timbortnik) [inline features/steps/step_selenium_api_token.py]: Please, do not copy-paste steps - reuse them https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

File: features/steps/step_selenium_api_token.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+@when('we edit profile')
+def step_impl(context):
+    context.authorized_page.edit_profile()
+    assert '/account' in context.authorized_page.current_url()
+
+
+@then('we click API access')
+def step_impl(context):
+    context.authorized_page.api_access()
```
Comment (timbortnik) [inline features/steps/step_selenium_api_token.py]: Should it be a different page object? Like api_page?

Comment (timbortnik) [reply features/steps/step_selenium_api_token.py]: Or profile_page?

**Review by timbortnik — CHANGES_REQUESTED** [review]

File: features/environment_secret.py (hunk 2 above)
Comment (Uarhodesian) [reply features/environment_secret.py]: remove password

File: features/steps/step_selenium_api_token.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: UTF-8 -*-
+
+# @mark.steps
+# ----------------------------------------------------------------------------
+# STEPS:
+# ----------------------------------------------------------------------------
+from behave import given, when, then
+import time
```
Comment (helltepakcer) [inline features/steps/step_selenium_api_token.py]: If you don't use import module - remove it please.

File: pages/authorized_page.py [hunk 6]
```diff
@@ -15,3 +15,28 @@ def label_page_head(self):
 
     def get_page_head(self):
         return self.label_page_head().text
+
```
Comment (helltepakcer) [inline pages/authorized_page.py]: All this logic should be in other file, this scenario don't happen in authorized page.

File: features/steps/step_selenium_api_token.py (hunk 4 above)
Comment (Uarhodesian) [reply features/steps/step_selenium_api_token.py]: done

File: features/api_token_settings.feature [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    And we enter password
+    Then we see Welcome title
+
+Scenario: Check create API token
+    Given we are on Account settings Page
+    When we click API access
+    When we reenter password
+    Then we are on API access Page
+    When we create new API token
+    Then we see new API token
```
Comment (timbortnik) [inline features/api_token_settings.feature]: I'd suggest to read why EOF without new line is considered a bad style https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

File: features/environment.py [hunk 8]
```diff
@@ -16,38 +16,25 @@
...
     context.base_url = "https://bortnik.hipchat.com"
     context.driver = webdriver.Chrome()
-
     context.login_page = LoginPage(context)
     context.authorized_page = AuthorizedPage(context)
+    context.api_page = ApiPage(context)
     context.settings_page = SettingsPage(context)
 
 
-def after_scenario(context, scenario):
```
Comment (timbortnik) [inline features/environment.py]: What is the reason to remove this code?

File: pages/api_page.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+    def check_token(self):
+        ch_token = self.context.driver.find_element_by_id('label').get_attribute('value')
+        if len(ch_token) > 0:
+            return True
+        else:
+            return False
+
+    def del_token(self):
+        pass
```
Comment (timbortnik) [inline pages/api_page.py]: Shall we actually delete a token here?

File: pages/authorized_page.py (hunk 6 above)
Comment (timbortnik) [reply pages/authorized_page.py]: current_url looks like a good candidate for base page

File: pages/settings_page.py [hunk 10]
```diff
@@ -19,6 +19,15 @@ def full_name(self):
     def email(self):
         return self.context.driver.find_element_by_id('email').get_attribute('value')
 
+    def current_url(self):
```
Comment (timbortnik) [inline pages/settings_page.py]: DRY https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

File: features/environment.py [hunk 11]
```diff
@@ -16,38 +16,25 @@
...
     context.base_url = "https://bortnik.hipchat.com"
     context.driver = webdriver.Chrome()
-
     context.login_page = LoginPage(context)
     context.authorized_page = AuthorizedPage(context)
+    context.api_page = ApiPage(context)
     context.settings_page = SettingsPage(context)
-
-
-def after_scenario(context, scenario):
```
Comment (timbortnik) [inline features/environment.py]: What is the reason to remove this code?

File: pages/api_page.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    def check_token_by_name(self, token_name):
+        for i in self.context.driver.find_elements_by_css_selector('#tokens tbody  tr.data'):
+            if i.find_element_by_css_selector('td span.editable').text == token_name:
+                return i.find_element_by_css_selector('td.token').text
+
+    def delete_token_by_name(self, token_name):
+        for i in self.context.driver.find_elements_by_css_selector('#tokens tbody  tr.data'):
+            if i.find_element_by_css_selector('td span.editable').text == token_name:
+                i.find_element_by_css_selector(' tr.data td form.aui').click()
+        alert = self.context.driver.switch_to_alert()
```
Comment (timbortnik) [inline pages/api_page.py]: switch_to.alert ?

**Review by timbortnik — APPROVED** [review]

**Pull request merged**